src/file-upload/file-item-edit.tsx

```diff
@@ -0,0 +1,162 @@
+    } = fileItem;
+
+    const [formattedName, setFormattedName] = useState<string>();
+    const [currentDescription, setCurrentDescription] = useState<string>("");
+
+    const textareaRef = useRef<HTMLTextAreaElement>();
+    const nameSectionRef = useRef<HTMLDivElement>();
+
+    // =========================================================================
+    // EFFECTS
+    // =========================================================================
+    useEffect(() => {
+        setFormattedName(getTruncatedText(name));
+    }, [wrapperWidth]);
+
+    useEffect(() => {
+        setCurrentDescription(fileItem.description || "");
+    }, [fileItem]);
+
+    // =========================================================================
+    // EVENT HANDLERS
+    // =========================================================================
+    const handleSave = () => {
+        onSave(textareaRef.current.value.trim());
+    };
+
+    const handleChange = (event: React.ChangeEvent<HTMLTextAreaElement>) => {
+        setCurrentDescription(event.target.value);
+    };
+
+    const handleBlur = (event: React.FocusEvent<HTMLTextAreaElement>) => {
+        onBlur(event.target.value);
+    };
+
+    // =========================================================================
+    // HELPER FUNCTIONS
+    // =========================================================================
+    const getTruncatedText = (value: string) => {
+        if (!truncateText) return value;
+
+        const widthOfElement =
+            nameSectionRef && nameSectionRef.current
+                ? nameSectionRef.current.getBoundingClientRect().width
+                : 0;
+
+        return StringHelper.truncateOneLine(
+            value,
+            widthOfElement,
+            widthOfElement / 2,
+            widthOfElement / 2 / 8, // Arbitrary
+            16 // Font size
+        );
+    };
+
+    const shouldDisableSave = () => {
+        const trimmedDescription = currentDescription.trim();
+        return trimmedDescription.length === 0;
+    };
+
+    // =========================================================================
+    // RENDER FUNCTIONS
+    // =========================================================================
+    const renderFileNameAndSize = () => (
+        <NameSection ref={nameSectionRef}>
+            <FileNameText weight="semibold">{formattedName}</FileNameText>
+            <FileSizeText>
+                {FileUploadHelper.formatFileSizeDisplay(size)}
+            </FileSizeText>
+        </NameSection>
+    );
+
+    return (
+        <Item data-testid={`${id}-edit-display`}>
+            <ContentSection>
+                {thumbnailImageDataUrl && (
+                    <FileListItemThumbnail
+                        thumbnailImageDataUrl={thumbnailImageDataUrl}
+                    />
+                )}
+                <DetailsSection>
+                    {renderFileNameAndSize()}
+                    <Form.Textarea
```

Comment:
missed out on previous reviews. does the textarea resizing need to be restricted?

Reply:
Oh yes! I seem to have some recollection talking about this. Let me clarify

Reply:
Clarified and Selig mentioned no need.